wasmplugin/plugin.go

```diff
@@ -365,3 +394,51 @@ func logError(error ctypes.MatchedRule) {
 		proxywasm.LogDebug(msg)
 	}
 }
+
+type AddressInfo struct {
+	IP   string
+	port int
+}
+
+// Retrieves adddress properties from the proxy
+// Expected targets are "source" or "destination"
+// Envoy ref: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes#connection-attributes
+func RetrieveAddressInfo(target string) AddressInfo {
+	var targetAddress AddressInfo
+	srcAddressRaw, err := proxywasm.GetProperty([]string{target, "address"})
+	if err != nil {
+		proxywasm.LogWarnf("failed to get %s address: %v", target, err)
+	} else {
+		targetAddress.IP, _, err = net.SplitHostPort(string(srcAddressRaw))
+		if err != nil {
+			proxywasm.LogWarnf("failed to parse %s address: %v", target, err)
+		}
+	}
+
+	srcPortRaw, err := proxywasm.GetProperty([]string{target, "port"})
+	if err != nil {
+		proxywasm.LogInfof("failed to get %s port: %v", target, err)
+	} else {
+		targetAddress.port, err = parsePort(srcPortRaw)
+		if err != nil {
+			proxywasm.LogWarnf("failed to parse %s port: %v", target, err)
+		}
+	}
+	return targetAddress
+}
+
+// Converts port, retrieved as little-endian bytes, into int
+func parsePort(b []byte) (int, error) {
+	// Port attribute ({"source", "port"}) is populated as uint64 (8 byte)
```

Comment:
```suggestion
	// Port attribute ({"source", "port"}) is populated as uint32 (4 byte)
```

Or at least that's how the code is working

Reply:
Testing it, Envoy returns 8 bytes. I was taking into account only 4 bytes because of Port limits specifics. Yet it may lead to misunderstandings. Using now all the 8 bytes received.